eppic-cli/src/main/java/eppic/assembly/Assembly.java

```diff
@@ -955,21 +974,64 @@ public String toString() {
 	}
 
 	/**
-	 * Assign a call to this Assembly, retrieve the call subsequently with {@link #getCall()}
+	 * Compute an initial (non-normalized) probabilistic score from
+	 * the individual interface probabilities.
 	 */
-	public void score() {
-
-		// we won't support non-fully-covering stoichiometries for the moment
-		// TODO support them: most likely requires a more complex data model where we can have score/calls for each of the subcomponents of the assembly
-		if (!assemblyGraph.isFullyCovering()) {
-			logger.warn("Assembly {} does not cover all entities, assembly scoring will be done for first sub-assembly only", toString());
-		}
+	public void calcScore() {
 		
+		// Construct a list of all the smaller powersets of this assembly that are equivalent
+		List<PowerSet> pss = new ArrayList<PowerSet>();
 		
-		SubAssembly firstSubAssembly = assemblyGraph.getSubAssemblies().get(0);
+		for (PowerSet ps : engagedSet.getOnPowerSet()) {
+			// Equivalent means that they have the same number of subassemblies
+			Assembly pa = new Assembly(crystalAssemblies, ps);
+			if (pa.getAssemblyGraph().getSubAssemblies().size() == 
+					this.getAssemblyGraph().getSubAssemblies().size())
+				pss.add(ps);
+		}
```

Comment:
The condition is needed because only those parents that form the same assembly have to be included in the probability of the assembly. This is checked by removing interfaces and seeing if the components of the graph are the same (removing edges can only increase the number of components, in which case that parent does not have to be included).

As an example, a dihedral assembly with interfaces 1,2 and 3, would have contributions from subsets (1,2), (2,3) and (1,3), but not from (1), (2) or (3), because those form dimers.

It is also a way to generically calculate the minimum number of interfaces required for an assembly to be formed. For homomeric dihedrals is 2 and for cyclics 1, but it gets more complicated for heteromers.